test.js

```diff
@@ -504,6 +504,30 @@ describe('EventEmitter', function tests() {
       assume(e.listeners('foo')).eql([]);
       assume(e._eventsCount).equals(0);
     });
+
+    it('removes a listener when removed as part handler callback', function() {
+      var e = new EventEmitter()
+        , foo = { bar: true }
+        , calledThird = 0;
+
+      function first() {
+        foo = null;
+        e.removeListener('update', second);
+      }
+      function second() {
```

Comment:
Copying this test what's in master (no change from PR), this test fails with `TypeError: Cannot read property 'bar' of null`. Current master attempts to call `second` in the same emit call. It's not removed until the next stack.